internal/bed/read.go

```diff
@@ -0,0 +1,84 @@
+package bed
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Opening and reading the bed file
+func (bf *Bedfile) Read() error {
+	file, err := os.Open(bf.Input)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	return bf.read(file)
+}
+
+// Reading the bed file
+func (bf *Bedfile) read(file io.Reader) error {
+	var err error
+	var expectedNrOfCols int
+
+	minNrCols := 3
+	headerPattern := regexp.MustCompile(`^(browser|track|#)`)
+	strandPattern := regexp.MustCompile(`^(\.|\+|-|\+1|-1|1)$`)
+
+	lineNr := 0
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var l Line
+		lineNr++
+
+		lineText := scanner.Text()
+
+		// Handle headers
+		if headerPattern.MatchString(lineText) && len(bf.Lines) == 0 {
+			bf.Header = append(bf.Header, lineText)
+			continue
+		}
+
+		// Split line
+		l.Full = strings.Split(lineText, "\t")
+
+		// For the first content line save the number of columns
+		if lineNr == len(bf.Header)+1 {
+			expectedNrOfCols = len(l.Full)
+			if expectedNrOfCols < minNrCols {
+				return fmt.Errorf("less than 3 columns on line %d: %s", lineNr, lineText)
+			}
+		}
+		if len(l.Full) != expectedNrOfCols {
+			return fmt.Errorf("expected %d columns on line %d got %d: %s",
+				expectedNrOfCols, lineNr, len(l.Full), lineText)
+		}
+
+		// Fill struct
+		l.Chr = l.Full[chrIdx]
+		l.Start, err = strconv.Atoi(l.Full[startIdx])
+		if err != nil {
+			return fmt.Errorf("non-int start position on line %d: %s", lineNr, l.Full[startIdx])
+		}
+		l.Stop, err = strconv.Atoi(l.Full[stopIdx])
+		if err != nil {
+			return fmt.Errorf("non-int stop position on line %d: %s", lineNr, l.Full[stopIdx])
+		}
+		// Verify strand and gene column info
+		if bf.StrandCol > stopIdx {
```

Comment:
Just wondering, you are checking this also in `VerifyAndHandle()`, do you need to assess that they (both `bf.StrandCol` and `bf.FeatCol`) are greating than again? 🤔

Reply:
Ah, I see I overlooked this conversation.
This seems to be a comment that is a bit outdated. 

The intention of this comparison is just to make sure we are not setting `StrandCol` or `FeatCol` if they were 0. 

While making this and the `VerifyAndHandle()` function I jumped a bit back and forth, so some things have been moved arround.

I have already merged, but I promise to update this comment in the next pull request 🙂

Reply:
Updated in 49999aa8f3a0866523945c88234197c5beb95bec

Hope this is more understandable?

Reply:
it is 👍